eth/handler.go

```diff
@@ -385,7 +385,7 @@ func (h *handler) runEthPeer(peer *eth.Peer, handler eth.Handler) error {
 		return errors.New("peer dropped during handling")
 	}
 	// Register the peer in the downloader. If the downloader considers it banned, we disconnect
-	if err := h.downloader.RegisterPeer(peer.ID(), peer.Version(), peer); err != nil {
```

Comment:
this just makes it more convenient, right? Doesn't seem to change anything

Reply:
I had to make this change otherwise the compiler will complain that `peer` didn't implement `RequestWitnesses`.

Reply:
Aha, okay.